R/geom-path.r

```diff
@@ -319,9 +319,19 @@ stairstep <- function(data, direction="hv") {
     xs <- c(1, rep(2:n, each = 2))
   }
 
-  data.frame(
-    x = data$x[xs],
-    y = data$y[ys],
-    data[xs, setdiff(names(data), c("x", "y"))]
-  )
+  if (direction == "hvh") {
+    gaps <- data$x[2:n] - data$x[1:(n-1)]
```

Comment:
I think this would be simpler as

```R
 gaps <- data$x[-1] - data$x[-n]
```

And might even better to divide by two here.

Reply:
Good point. Instead of dividing by 2 here, I created a new variable that creates a vector of the midpoints by adding gaps/2 to the prior data (excepting the final datapoint).
```
gaps <- data$x[-1] - data$x[-n] 
mid_x <- data$x[-n] + gaps/2
```